msg.go

```diff
@@ -364,11 +367,21 @@ Loop:
 						for _, r := range bufs {
 							s = append(s, r)
 						}
+						maxLen += 3 // unprintable char is 3 extra bytes
 					} else {
 						s = append(s, b)
 					}
 				}
 			}
+			// never exceed the allowed label count lenght (63) 
+			if labels >= 63 { 
+				return "", lenmsg, &Error{err: "name exceeds 63 labels"}
+			}
+			labels += 1
+			// never exceed the allowed doman name length (255 octets) 
+			if len(s) >= maxLen {
+				return "", lenmsg, &Error{err: "name exceeded allowed 255 octets"}
```

Comment:
You might also consider special treatment for names ending with the "`.local`" TLD, since [RFC 6762](https://tools.ietf.org/html/rfc6762#appendix-C) defines those as link-local and allows them to occupy 256 octets in wire format for mDNS.